### PR TITLE
Feature/cleaning dataset

### PR DESCRIPTION
**Pull request by javiermediavilla**
Primer intento de pull request

### CODE REVIEW

File: DatasetCleaning.py [hunk 1]
```diff
@@ -1,99 +1,210 @@
-# -*- coding: utf-8 -*-
 """
 Created on Wed Mar 17 09:43:28 2021
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:2]: Test comentario

Comment (javiermediavilla) [reply DatasetCleaning.py:2]: No estoy deacuerdo

Comment (mgarciafernandez-altran) [reply DatasetCleaning.py:2]: Vale haz lo que quieras

**Comment by mgarciafernandez-altran** [comment]
Esto esta todo mal

File: DatasetCleaning.py [hunk 2]
```diff
@@ -1,99 +1,210 @@
...
 
 
-El objetivo de este stript de Python es generar una versión preprocesada del dataset.
+Main objetive:
+The main objetive of the script is to generate a preprocessed version o the dataset 'crimes_dataset.csv'
+and then save a copy in numpy format. 
+The rest of the work during the project will use these numpy version of the dataset.
 """
 
+#### Packages ####
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:13]: General rules about importing libraries and packages at Python:
1.- Never ever do `import X as Y`
2.- `import X` is preferred over `from X import Y`
3.- Libraries shall be dividen on three grous ordered: standard libraries, thrid party libraries, custom libraries.
4.- inside each group libraries shall be in alphabetical order

File: DatasetCleaning.py [hunk 3]
```diff
@@ -1,99 +1,210 @@
-# -*- coding: utf-8 -*-
 """
 Created on Wed Mar 17 09:43:28 2021
 
 @author: javim
 
 
-El objetivo de este stript de Python es generar una versión preprocesada del dataset.
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:8]: Comments always in English

File: DatasetCleaning.py [hunk 4]
```diff
@@ -1,99 +1,210 @@
...
+x_train_standard      = (x_train-mean_standarization)/std_standarization
+x_test_standard       = (x_test-mean_standarization)/std_standarization
+##############################
+
+
+######## SAVING THE DATA SETS ########
+if save_data == True:
+    save_path = actual_path + '/Preprocessed_Dataset/' + file_save_dataset
+    np.savez(save_path, x_train = x_train_standard, x_test = x_test_standard, y_train = y_train, y_test = y_test, mean_standarization = mean_standarization, std_standarization = std_standarization)
+######################################
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:209]: General comment on every file. If the file is intended to be 'executable', that is run with `python file.py` or even `./file.py` it MUST have defined a main function `def main()` and a `if __name__ == '__main__':` sencetence

File: DatasetCleaning.py [hunk 5]
```diff
@@ -1,99 +1,210 @@
...
-    dump(RF_model, file_save_model) 
-    print('\nModelo Guardando')
-else:
-    print('\nModelo NO guardado. Poner variable "guardado_modelo = True" para guardarlo')
-##############################################
+for ii in range(len(data_numpy)):
+    policial_resources[ii] = cost_policy[np.where(ofense_codes == data['OFFENSE_CODE'][ii])[0][0]] 
     
+slot = np.zeros(len(data_numpy))
+for ii in range(len(data_numpy)):
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:144]: Give meaningful names to all the variables, even if it is an index. In general, indexes have a functional name for example, if it is a flight id, even though it is an integer, shall be named something like `flight_id`. If it is something meaningless like the index of the list shall be named like that `list_index`.

File: DatasetCleaning.py [hunk 6]
```diff
@@ -1,99 +1,210 @@
...
+
+    word_time      = str(data['YEAR'][ii]) + str(data['MONTH'][ii]) + str(data['DAY_OF_WEEK'][ii]) + str(data['DAY_OF_THE_MONTH'][ii])
+    word_location  = str(data['REPORTING_AREA'][ii])
+    slot[ii]  = float(word_time+word_location)
+
+slot_sort             = np.sort(slot)
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:150]: In general we follow the PEP-8 rules. This means that lines below shall be something like `unique_slots = len...`. That is the symbol `=` shall be enclosed between single blank spaces

File: DatasetCleaning.py [hunk 7]
```diff
@@ -1,99 +1,210 @@
...
+ind_Datos           = range(len(data_X))
+ind_Datos_shuffle   = np.random.choice(ind_Datos,len(data_X), replace=False)
+
+data_X_shuffle      = data_X[ind_Datos_shuffle]
+data_Y_shuffle      = data_Y[ind_Datos_shuffle]
+########################################
+
+
+####### TRAIN/TEST SPLIT #######
+x_train, x_test, y_train, y_test = train_test_split(data_X_shuffle, data_Y_shuffle, test_size=0.25, random_state=4)
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:193]: As a rule, une must ALWAYS state the name of the arguments of functions when calling them, for example, `train_test_split(data_X_shuffle...` shall be `train_test_split(name_of_var=data_X...`

File: DatasetCleaning.py [hunk 8]
```diff
@@ -1,99 +1,210 @@
-# -*- coding: utf-8 -*-
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:1]: This first line of comment is junk added by IDE. It means nothing

File: DatasetCleaning.py [hunk 9]
```diff
@@ -1,99 +1,210 @@
...
 """
 Created on Wed Mar 17 09:43:28 2021
 
 @author: javim
 
 
-El objetivo de este stript de Python es generar una versión preprocesada del dataset.
+Main objetive:
+The main objetive of the script is to generate a preprocessed version o the dataset 'crimes_dataset.csv'
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:8]: Main comment of file shall be more descriptive

File: DatasetCleaning.py [hunk 10]
```diff
@@ -1,99 +1,210 @@
...
 """
-Para empezar, se va a hacer una asignación aleatoria de los recursos policiales 
-necesarios para gestionar cada incidente, posteriormente se puede usar una 
-polçitica de costes más realista
+Lets convert the variables DAY_OF_WEEK and UCR_PART from string into integer
 """
 
-np.random()
+mymap  = {'Monday':1, 'Tuesday':2, 'Wednesday':3, 'Thursday':4, 'Friday':5, 'Saturday':6, 'Sunday':7}
+data   = data.applymap(lambda s: mymap.get(s) if s in mymap else s)
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:118]: To not use the `lambda` directive. It complicates code review by external people and in general leads to bad programming practices. it is demonstrated that `lambda` sencences my be converted to more pythonic sentences or even to specific functions

File: DatasetCleaning.py [hunk 11]
```diff
@@ -1,99 +1,210 @@
...
+from scipy import stats
+from sklearn.model_selection import train_test_split
+##################
 
 
-print('\nPreprocesado Dataset...')
-#### Importamos los datos ####
-ficheroDatos  ='crimes_dataset.csv'
-separador     = ';'
+#### Load the data from csv ####
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:26]: In general on this script I miss more modularity There is no single function!!!

File: DatasetCleaning.py [hunk 12]
```diff
@@ -1,99 +1,210 @@
...
-else:
-    print('\nModelo NO guardado. Poner variable "guardado_modelo = True" para guardarlo')
-##############################################
+for ii in range(len(data_numpy)):
+    policial_resources[ii] = cost_policy[np.where(ofense_codes == data['OFFENSE_CODE'][ii])[0][0]] 
     
+slot = np.zeros(len(data_numpy))
+for ii in range(len(data_numpy)):
+
+    word_time      = str(data['YEAR'][ii]) + str(data['MONTH'][ii]) + str(data['DAY_OF_WEEK'][ii]) + str(data['DAY_OF_THE_MONTH'][ii])
```
Comment (mgarciafernandez-altran) [inline DatasetCleaning.py:146]: Do not use string contatenation! Use f-strings!! (see python official documentation)

**Review by mgarciafernandez-altran — CHANGES_REQUESTED** [review]

**Review by mgarciafernandez-altran — APPROVED** [review]

**Pull request merged**